lib/puppet/provider/package/powershellcore.rb

```diff
@@ -44,6 +44,32 @@ def update
     self.class.invoke_ps_command update_command
   end
 
+  # Turns a array of install_options into flags to be passed to a command.
+  # The options can be passed as a string or hash. Note that passing a hash
+  # should only be used in case "-foo bar" must be passed,
+  # Regular flags like '-foobar' must be passed as a string.
+  # which can be accomplished with:
+  #     install_options => [ '-foobar',{ '-foo' => 'bar' } ]
+  # This will result in the following being passed as arguments to the command:
+  #     -foobar -foo bar
+  # @param options [Array]
+  # @return Concatenated list of options
+  # @api private
+  def install_options
+    return unless @resource[:install_options]
+
+    @resource[:install_options].collect do |val|
```

Comment:
Could we make this a param rather than reading straight from the resource? It makes for easier testing
```ruby
def install_options(options)
    options.collect do |val|
```